packages/flutter_tools/lib/src/vscode/vscode.dart

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/platform.dart';
+import '../base/version.dart';
+
+// VS Code layout:
+
+// macOS:
+//   /Applications/Visual Studio Code.app/Contents/
+//   /Applications/Visual Studio Code - Insiders.app/Contents/
+//   $HOME/Applications/Visual Studio Code.app/Contents/
+//   $HOME/Applications/Visual Studio Code - Insiders.app/Contents/
+// macOS Extensions:
+//   $HOME/.vscode/extensions
+//   $HOME/.vscode-insiders/extensions
+
+// Windows:
+//   $programfiles(x86)\Microsoft VS Code
+//   $programfiles(x86)\Microsoft VS Code Insiders
+// TODO: Confirm these are correct for 64bit
+//   $programfiles\Microsoft VS Code
+//   $programfiles\Microsoft VS Code Insiders
+// Windows Extensions:
+//   $HOME/.vscode/extensions
+//   $HOME/.vscode-insiders/extensions
+
+// Linux:
+//   /usr/share/code/bin/code
+//   /usr/share/code-insiders/bin/code-insiders
+// Linux Extensions:
+//   $HOME/.vscode/extensions
+//   $HOME/.vscode-insiders/extensions
+
+const String extensionIdentifier = 'Dart-Code.dart-code';
+
+const bool _includeInsiders =
+    false; // Include VS Code insiders (useful for debugging).
+
+class VsCode {
+  VsCode(this.directory, this.dataFolderName, {Version version})
```

Comment:
nit: I believe flutter styles wants spaces inside the curlies: `{ Version version }`

Reply:
Done in 58549d8 (it actually came from android_studio which has none, but seems like most of other code does have spaces so I added them).